File: app/src/main/java/de/bitb/astroskop/model/converter/IntegerListConverter.java
package de.bitb.astroskop.model.converter;

import android.text.TextUtils;

import java.util.ArrayList;
import java.util.List;

import io.objectbox.converter.PropertyConverter;

public class IntegerListConverter implements PropertyConverter<List<Integer>, String> {

    private static final java.lang.String SEPERATOR = ";";

    @Override
    public List<Integer> convertToEntityProperty(String databaseValue) {
        String[] split = databaseValue.split(SEPERATOR);
        List<Integer> result = new ArrayList<>();
        for (String aSplit : split) {
            result.add(Integer.valueOf(aSplit));
        }
        return result;
    }

    @Override
    public String convertToDatabaseValue(List<Integer> integers) {
        String result = "";
        for (Integer integer : integers) {
            if (!TextUtils.isEmpty(result)) {
                result += SEPERATOR;
            }
            result += integer;
        }
        return result;
    }

}

File: app/src/main/java/de/bitb/astroskop/model/converter/IntegerListConverter.kt
package de.bitb.astroskop.model.converter

import android.text.TextUtils

import java.util.ArrayList

import io.objectbox.converter.PropertyConverter

class IntegerListConverter : PropertyConverter<List<Int>, String> {

    override fun convertToEntityProperty(databaseValue: String): List<Int> {
        val split = databaseValue.split(SEPERATOR.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val result = ArrayList<Int>()
        for (aSplit in split) {
            result.add(Integer.valueOf(aSplit))
        }
        return result
    }

    override fun convertToDatabaseValue(integers: List<Int>): String {
        var result = ""
        for (integer in integers) {
            if (!TextUtils.isEmpty(result)) {
                result += SEPERATOR
            }
            result += integer
        }
        return result
    }

    companion object {

        private val SEPERATOR = ";"
    }

}
